### PR TITLE
fix host network resolver issue with alpine

### LINKED ISSUE #1172: vz: host resolver not working
### Description

I noticed that the host resolver is not working with the vz driver https://github.com/lima-vm/lima/pull/1147, and it is not documented among the known issues.

### PR DESCRIPTION
**Pull request by balajiv113**
fixes #1172

### CODE REVIEW

File: pkg/networks/gvisor.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func StartGVisorNetstack(ctx context.Context, gVisorOpts *GVisorNetstackOpts) {
 		Debug:             false,
 		MTU:               opts.MTU,
 		Subnet:            SlirpNetwork,
-		GatewayIP:         SlirpGateway,
+		GatewayIP:         SlirpDNS,
```
Comment (AkihiroSuda) [inline pkg/networks/gvisor.go:46]: Could you add a code comment to explain this

Comment (balajiv113) [reply pkg/networks/gvisor.go:47]: Done. Hope this is clear now.

**Review by abiosoft — DISMISSED** [review]
This works 🎉 

Thanks once again @balajiv113

**Comment by balajiv113** [comment]
Modified comments with some more info :) To keep it informative

**Review by abiosoft — DISMISSED** [review]

File: pkg/networks/gvisor.go (hunk 1 above)
Comment (abiosoft) [reply pkg/networks/gvisor.go:47]: The convention in Go for code comments is to have a space between the double slash `//` and the text.

Comment (jandubois) [reply pkg/networks/gvisor.go:47]: I have to admit that even with the comment, I still don't understand **why** this works. Why is it called a gateway if it isn't actually a gateway?

Comment (abiosoft) [reply pkg/networks/gvisor.go:47]: I think the constants specific to vz driver should be re-defined to avoid confusion. Type aliases can be used.

Comment (AkihiroSuda) [reply pkg/networks/gvisor.go:47]: Yes, thanks, LGTM when the above is added as code comment lines

Comment (BalajiVijayakumar1195) [reply pkg/networks/gvisor.go:52]: Done

Comment (balajiv113) [reply pkg/networks/gvisor.go:52]: Done

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**